### PR TITLE
Added Abstract Ellipse Rev1

### PR DESCRIPTION
**Pull request by AbstractKB**
## Description

Added the Abstract keyboards folder, the Ellipse folder, and the Rev 1 of the Ellipse.  Also added 2 keymaps for the Ellipse, default and another tester keymap called abstractkb.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/abstract/ellipse/keymaps/abstractkb/config.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/* Copyright 2018 REPLACE_WITH_YOUR_NAME
```
Comment (fauxpark) [inline keyboards/abstract/ellipse/keymaps/abstractkb/config.h]:
```suggestion
/* Copyright 2019 AbstractKB
```

File: keyboards/abstract/ellipse/keymaps/abstractkb/keymap.c [hunk 2]
```diff
@@ -0,0 +1,66 @@
+/* Copyright 2018 AbstractKB
```
Comment (fauxpark) [inline keyboards/abstract/ellipse/keymaps/abstractkb/keymap.c]:
```suggestion
/* Copyright 2019 AbstractKB
```

File: keyboards/abstract/ellipse/keymaps/abstractkb/keymap.c [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+  MYKEY = SAFE_RANGE
```
Comment (fauxpark) [inline keyboards/abstract/ellipse/keymaps/abstractkb/keymap.c]: This keycode doesn't appear to be used.

File: keyboards/abstract/ellipse/keymaps/abstractkb/keymap.c [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT( /* Base */
+    KC_MUTE,  RGB_TOG,  BL_TOGG, \
```
Comment (fauxpark) [inline keyboards/abstract/ellipse/keymaps/abstractkb/keymap.c]:
```suggestion
    KC_MUTE,  RGB_TOG,  BL_TOGG,
```

File: keyboards/abstract/ellipse/keymaps/abstractkb/keymap.c [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    RGB_M_SW,  RGB_M_P,	BL_BRTG \
```
Comment (fauxpark) [inline keyboards/abstract/ellipse/keymaps/abstractkb/keymap.c]:
```suggestion
    RGB_M_SW,  RGB_M_P,	BL_BRTG
```

File: keyboards/abstract/ellipse/keymaps/default/config.h [hunk 6]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2018 REPLACE_WITH_YOUR_NAME
```
Comment (fauxpark) [inline keyboards/abstract/ellipse/keymaps/default/config.h]:
```suggestion
/* Copyright 2019 AbstractKB
```

File: keyboards/abstract/ellipse/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,66 @@
+/* Copyright 2018 AbstractKB
```
Comment (fauxpark) [inline keyboards/abstract/ellipse/keymaps/default/keymap.c]:
```suggestion
/* Copyright 2019 AbstractKB
```

File: keyboards/abstract/ellipse/rev1/config.h [hunk 8]
```diff
@@ -0,0 +1,245 @@
+/*
+Copyright 2018 AbstractKB
```
Comment (fauxpark) [inline keyboards/abstract/ellipse/rev1/config.h]:
```suggestion
Copyright 2019 AbstractKB
```

File: keyboards/abstract/ellipse/rev1/config.h [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+ * the keyboard. They are best used in combination with the HID Listen program,
+ * found here: https://www.pjrc.com/teensy/hid_listen.html
+ *
+ * The options below allow the magic key functionality to be changed. This is
+ * useful if your keyboard/keypad is missing keys and you want magic key support.
+ *
+ */
+
+/* key combination for magic key command */
+#define IS_COMMAND() ( \
```
Comment (fauxpark) [inline keyboards/abstract/ellipse/rev1/config.h]: Please remove this block, as it's already defined in QMK core, and you haven't changed the default.

File: keyboards/abstract/ellipse/rev1/rev1.c [hunk 10]
```diff
@@ -0,0 +1,43 @@
+/* Copyright 2018 AbstractKB
```
Comment (fauxpark) [inline keyboards/abstract/ellipse/rev1/rev1.c]:
```suggestion
/* Copyright 2019 AbstractKB
```

File: keyboards/abstract/ellipse/rev1/rev1.h [hunk 11]
```diff
@@ -0,0 +1,36 @@
+/* Copyright 2018 AbstractKB
```
Comment (fauxpark) [inline keyboards/abstract/ellipse/rev1/rev1.h]:
```suggestion
/* Copyright 2019 AbstractKB
```

File: keyboards/abstract/ellipse/rev1/rules.mk [hunk 12]
```diff
@@ -0,0 +1,82 @@
+# MCU name
+#MCU = at90usb1286
```
Comment (fauxpark) [inline keyboards/abstract/ellipse/rev1/rules.mk]:
```suggestion
```

File: keyboards/abstract/ellipse/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+#include QMK_KEYBOARD_H
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+  MYKEY = SAFE_RANGE
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT( /* Base */
+    KC_A,	RGB_TOG,	KC_C, \
```
Comment (fauxpark) [inline keyboards/abstract/ellipse/keymaps/default/keymap.c]:
```suggestion
    KC_A,	RGB_TOG,	KC_C,
```

File: keyboards/abstract/ellipse/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+    KC_X,	KC_Y,		KC_Z \
```
Comment (fauxpark) [inline keyboards/abstract/ellipse/keymaps/default/keymap.c]:
```suggestion
    KC_X,	KC_Y,		KC_Z
```

File: keyboards/abstract/ellipse/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+  MYKEY = SAFE_RANGE
```
Comment (noroadsleft) [inline keyboards/abstract/ellipse/keymaps/default/keymap.c]: You're not using this keycode in this keymap either.

File: keyboards/abstract/ellipse/readme.md [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+
+![EllipseRev1](http://abstractkb.tk/wp-content/uploads/2019/03/ellipsetop.jpg)
+
+A small keypad with more knobs than ever before!  3 Knobs and 3 MX-style switches on one little board.
+
+At this time there are 2 small known issues with the Rev 1.  One of which is that backlight breathing does not work,
+and the other issue that has been encountered is that the post_init_user function does not seem to be called.
+Hopefully these issues can be resolved soon; this QMK file is being released as orders have started to ship and they need to be configurable.
+
+Keyboard Maintainer: [AbstractKB](https://github.com/abstractkb)
```
Comment (noroadsleft) [inline keyboards/abstract/ellipse/readme.md]: Add two spaces here...

```suggestion
Keyboard Maintainer: [AbstractKB](https://github.com/abstractkb)  
```

File: keyboards/abstract/ellipse/readme.md [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+
```
Comment (noroadsleft) [inline keyboards/abstract/ellipse/readme.md]: Delete this line...

```suggestion
```

File: keyboards/abstract/ellipse/readme.md [hunk 18]
```diff
@@ -0,0 +1,21 @@
...
+Hardware Supported: The Abstract Ellipse Rev1
```
Comment (noroadsleft) [inline keyboards/abstract/ellipse/readme.md]: Two more spaces here...

```suggestion
Hardware Supported: The Abstract Ellipse Rev1  
```

File: keyboards/abstract/ellipse/readme.md [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+
```
Comment (noroadsleft) [inline keyboards/abstract/ellipse/readme.md]: ... and delete this line, too.

```suggestion
```

File: keyboards/abstract/ellipse/rev1/config.h [hunk 20]
```diff
@@ -0,0 +1,245 @@
...
+*/
+
+/* Bootmagic Lite key configuration */
+// #define BOOTMAGIC_LITE_ROW 0
+// #define BOOTMAGIC_LITE_COLUMN 0
+
+#define NUMBER_OF_ENCODERS 3
+#define ENCODERS_PAD_A { B2, B3, D5 }
+#define ENCODERS_PAD_B { B1, B7, B4 }
+#define ENCODER_RESOLUTION 2
```
Comment (drashna) [inline keyboards/abstract/ellipse/rev1/config.h]: Could you make sure there is a blank line at the end here, for formatting reasons? 
```suggestion
#define ENCODER_RESOLUTION 2

```

File: keyboards/abstract/ellipse/rev1/rev1.c [hunk 21]
```diff
@@ -0,0 +1,43 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "rev1.h"
+
+void matrix_init_kb(void) {
```
Comment (drashna) [inline keyboards/abstract/ellipse/rev1/rev1.c]: If all of these are set to the defaults, they're not actually needed.

File: keyboards/abstract/ellipse/rev1/rules.mk [hunk 22]
```diff
@@ -0,0 +1,81 @@
...
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no      # Virtual DIP switch configuration(+1000)
```
Comment (drashna) [inline keyboards/abstract/ellipse/rev1/rules.mk]: Does the keyboard have a physical reset button, and one that's easy to get to? 

If not, you may want to consider enabling this.
```suggestion
BOOTMAGIC_ENABLE = lite
```

Comment (bengw99) [reply keyboards/abstract/ellipse/rev1/rules.mk]: yep, physical reset button and large hole to be able to access it.  Will fix the other stuff later today.

Comment (AbstractKB) [reply keyboards/abstract/ellipse/rev1/rules.mk]: yeah, has a reset button and a large hole to be able to access it.  Will fix the other stuff later today

Comment (drashna) [reply keyboards/abstract/ellipse/rev1/rules.mk]: Okay. Just wanted to make sure. :)

**Comment by drashna** [comment]
Thanks!

**Pull request merged**